### PR TITLE
frontDoorManagedIdentity

### PR DESCRIPTION
**Pull request by fatima99s**

### CODE REVIEW

File: plugins/azure/frontdoor/frontDoorManagedIdentity.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Front Door Managed Identity Enabled',
+    category: 'Front Door',
+    domain: 'Content Delivery',
+    severity: 'Medium',
+    description: 'Ensures that Azure Front Door have managed identity enabled.',
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorManagedIdentity.js:9]: can we write it as description: 'Ensures that Front Door profiles have managed identity enabled.',
Comment

File: plugins/azure/frontdoor/frontDoorManagedIdentity.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    more_info: 'Enabling managed identities eliminate the need for developers having to manage credentials by providing an identity for the Azure resource in Azure AD and using it to obtain Azure Active Directory (Azure AD) tokens.',
+    recommended_action: 'Modify the Front Door classic profile and add managed identity.',
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorManagedIdentity.js:11]: If its only available for classic profiles we should also mention this in description

File: plugins/azure/frontdoor/frontDoorManagedIdentity.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+            if (!profiles.data.length) {
+                helpers.addResult(results, 0, 'No existing Azure Front Door profiles found', location);
+                return rcb();
+            }
+
+            var frontDoorProfile = false;
+            profiles.data.forEach(function(profile) {
+                if (!profile.id || profile.kind!='frontdoor') return;
+
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorManagedIdentity.js:40]: According to the https://learn.microsoft.com/en-us/azure/frontdoor/managed-identity#prerequisites 
Before you can set up managed identity for Azure Front Door, you must have an Azure Front Door Standard or Premium profile created

can we also add check for premium and standard?

**Review by alphadev4 — CHANGES_REQUESTED** [review]

File: plugins/azure/frontdoor/frontDoorManagedIdentity.js (hunk 3 above)
Comment (fatima99s) [reply plugins/azure/frontdoor/frontDoorManagedIdentity.js:40]: in the list call we only get standard and premium forntdoor

File: plugins/azure/frontdoor/frontDoorManagedIdentity.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Front Door Managed Identity Enabled',
+    category: 'Front Door',
+    domain: 'Content Delivery',
+    severity: 'Medium',
+    description: 'Ensures that Azure Front Door Standard and Premium profile have managed identity enabled.',
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorManagedIdentity.js:9]:
```suggestion
    description: 'Ensures that Azure Front Door Standard and Premium profiles have managed identity enabled.',
```

File: plugins/azure/frontdoor/frontDoorManagedIdentity.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Front Door Managed Identity Enabled',
+    category: 'Front Door',
+    domain: 'Content Delivery',
+    severity: 'Medium',
+    description: 'Ensures that Azure Front Door Standard and Premium profiles have managed identity enabled.',
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorManagedIdentity.js:9]:
```suggestion
    description: 'Ensures that Azure Front Door standard and premium profiles have managed identity enabled.',
```

File: plugins/azure/frontdoor/frontDoorManagedIdentity.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    more_info: 'Enabling managed identities eliminate the need for developers having to manage credentials by providing an identity for the Azure resource in Azure AD and using it to obtain Azure Active Directory (Azure AD) tokens.',
+    recommended_action: 'Modify the Front Door Standard and Premium profile and add managed identity.',
```
Comment (alphadev4) [inline plugins/azure/frontdoor/frontDoorManagedIdentity.js:11]:
```suggestion
    recommended_action: 'Modify the Front Door standard and premium profile and add managed identity.',
```

**Review by alphadev4 — APPROVED** [review]

**Pull request merged**